lib/client.js

```diff
@@ -647,12 +647,14 @@ MqttClient.prototype.reconnect = function () {
   var f = function () {
     that.disconnecting = false
     that.disconnected = false
-    that.deferredReconnect = null
+    that._deferredReconnect = null
+    that.incomingStore = new that.incomingStore.constructor(that.incomingStore.options)
+    that.outgoingStore = new that.outgoingStore.constructor(that.outgoingStore.options)
```

Comment:
This constructor calling might be wrong. It assumes that `incomingStore` and `outgoingStore` have `options`, and their constructor has one parameter `options`.

I considered two more approaches.

1. Add a new interface like `resetInglights` to `Store`. And client calls it on `reconnect()`.
2. Add `reconnect()` method restriction. It can be called only if client is created with the option `clean: false`. User knows `clean` setting. If user want to reconnect with `clean: true`, user can call new `connect()`. That means creating the new connection but it is ok for `clean: true`.

What do you think?

Reply:
Approach 1 breaks existing user custom `Store` implementation. So Approach 2 is good. I will remove explicit constuctor calling code and add reconnect tests for both `clean: true` and `clean: false`.

Reply:
I updated the code using Approach 2.